test/e2e/specs/editor/blocks/heading.spec.js

```diff
@@ -179,4 +179,95 @@ test.describe( 'Heading', () => {
 			},
 		] );
 	} );
+
+	test( 'should change heading level with keyboard shortcuts', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '## Heading' );
+
+		// Change text alignment
+		await editor.clickBlockToolbarButton( 'Align text' );
+		const textAlignButton = page.locator(
+			'role=menuitemradio[name="Align text center"i]'
+		);
+		await textAlignButton.click();
+
+		// Focus the block content
+		await page.keyboard.press( 'Tab' );
+
+		await pageUtils.pressKeyWithModifier( 'access', '4' );
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/heading',
+				attributes: {
+					content: 'Heading',
+					textAlign: 'center',
+					level: 4,
+				},
+			},
+		] );
+	} );
+
+	test( 'should be converted from a paragraph to a heading with keyboard shortcuts', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'Paragraph' );
+
+		// Change text alignment
+		await editor.clickBlockToolbarButton( 'Align text' );
+		const textAlignButton = page.locator(
+			'role=menuitemradio[name="Align text center"i]'
+		);
+		await textAlignButton.click();
+
+		// Focus the block content
+		await page.keyboard.press( 'Tab' );
+
+		await pageUtils.pressKeyWithModifier( 'access', '2' );
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/heading',
+				attributes: {
+					content: 'Paragraph',
+					textAlign: 'center',
+					level: 2,
+				},
+			},
+		] );
+	} );
+
+	test( 'should be converted from a heading to a paragraph with keyboard shortcuts', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '## Heading' );
+
+		// Change text alignment
+		await editor.clickBlockToolbarButton( 'Align text' );
+		const textAlignButton = page.locator(
+			'role=menuitemradio[name="Align text center"i]'
+		);
+
+		await textAlignButton.click();
+
+		// Focus the block content
+		await page.keyboard.press( 'Tab' );
+
+		await pageUtils.pressKeyWithModifier( 'access', '0' );
+
+		// Somehow, if we use toMatchObject, it fails with GitHub Actions.
+		await expect.poll( editor.getEditedPostContent ).toBe(
+			`<!-- wp:paragraph {\"align\":\"center\"} -->
+<p class=\"has-text-align-center\">Heading</p>
+<!-- /wp:paragraph -->`
```

Comment:
Very strange, but only this test fails if `toMatchObject` is used. But locally it always succeeds...

Reply:
Oops! That's my bad! I recently changed the implementation of `editor.getBlocks` and it's causing this error. Rebasing should be able to reproduce this error. I'll put up another PR to fix that though! 🙇

Reply:
Proposed a fix in https://github.com/WordPress/gutenberg/pull/48093.

Reply:
Thanks for the investigation! I am checking to see if the test by toMatchObject passes correctly after temporarily applying the changes in #48093 to this PR.